# Data Overview

This notebook inspects the raw data: events and match information.

We will:
- Load and examine the structure of event data
- Load and examine match information
- Perform basic sanity checks on coordinates and event types
- Identify any data quality issues


In [5]:
%load_ext autoreload
%autoreload 2

import sys
from pathlib import Path

# Add project root to path
project_root = Path().resolve().parent
sys.path.insert(0, str(project_root))

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Import project modules
from src.data_loader import load_raw_events, load_match_info
from src.feature_engineering import add_zone_columns

# Set plotting style
plt.style.use("seaborn-v0_8")
sns.set_palette("husl")


The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


## Load Raw Data


In [6]:
# Load event data
events = load_raw_events()
print(f"Events shape: {events.shape}")
print(f"\nColumns: {list(events.columns)}")
print("\nFirst few rows:")
events.head()


Events shape: (579306, 18)

Columns: ['game_id', 'action_id', 'period_id', 'time_seconds', 'team_id', 'player_id', 'result_name', 'start_x', 'start_y', 'end_x', 'end_y', 'dx', 'dy', 'type_name', 'player_name_ko', 'team_name_ko', 'position_name', 'main_position']

First few rows:


game_id  action_id  period_id  time_seconds  team_id  player_id  \
0   126288          0          1         1.033     2353   356625.0   
1   126288          1          1         2.433     2353   356612.0   
2   126288          2          1         3.033     2353   356612.0   
3   126288          3          1         4.333     2353   356618.0   
4   126288          4          1         4.900     2353   356618.0   

  result_name    start_x    start_y      end_x      end_y         dx  \
0  Successful  52.670205  34.919632  68.628945  34.347412  15.958740   
1         NaN  68.628945  34.347412  68.628945  34.347412   0.000000   
2  Successful  68.228055  33.853256  65.468865  26.640292  -2.759190   
3         NaN  65.468865  26.640292  65.468865  26.640292   0.000000   
4  Successful  64.838025  26.766636  63.300510  43.018024  -1.537515   

          dy      type_name player_name_ko team_name_ko position_name  \
0  -0.572220           Pass            이영준  김천 상무 프로축구단            CF   
1   0.000000  Pass Received            원두재  김천 상무 프로축구단           CDM   
2  -7.212964           Pass            원두재  김천 상무 프로축구단           CDM   
3   0.000000  Pass Received            김진규  김천 상무 프로축구단            CM   
4  16.251388           Pass            김진규  김천 상무 프로축구단            CM   

  main_position  
0            CF  
1            CM  
2            CM  
3            CM  
4            CM

In [7]:
# Load match information
match_info = load_match_info()
print(f"Match info shape: {match_info.shape}")
print(f"\nColumns: {list(match_info.columns)}")
print("\nFirst few rows:")
match_info.head()


Match info shape: (198, 17)

Columns: ['game_id', 'season_id', 'competition_id', 'game_day', 'game_date', 'home_team_id', 'away_team_id', 'home_score', 'away_score', 'venue', 'competition_name', 'country_name', 'season_name', 'home_team_name', 'home_team_name_ko', 'away_team_name', 'away_team_name_ko']

First few rows:


game_id  season_id  competition_id  game_day            game_date  \
0   126283       3669             587         1  2024-03-01 05:00:00   
1   126284       3669             587         1  2024-03-01 07:30:00   
2   126285       3669             587         1  2024-03-02 05:00:00   
3   126286       3669             587         1  2024-03-02 07:30:00   
4   126287       3669             587         1  2024-03-02 07:30:00   

   home_team_id  away_team_id  home_score  away_score         venue  \
0          2354          4639           1           0      문수월드컵경기장   
1          4640          4657           1           1      전주월드컵경기장   
2          4648           316           2           0        광주전용구장   
3          4646          4220           0           1     인천축구전용경기장   
4          4643          4641           1           1  송암스포츠타운 주경기장   

  competition_name country_name  season_name          home_team_name  \
0       K League 1           KR         2024             Ulsan HD FC   
1       K League 1           KR         2024  Jeonbuk Hyundai Motors   
2       K League 1           KR         2024              Gwangju FC   
3       K League 1           KR         2024          Incheon United   
4       K League 1           KR         2024              Gangwon FC   

  home_team_name_ko        away_team_name away_team_name_ko  
0          울산 HD FC       Pohang Steelers           포항 스틸러스  
1         전북 현대 모터스  Daejeon Hana Citizen         대전 하나 시티즌  
2              광주FC              FC Seoul              FC서울  
3          인천 유나이티드              Suwon FC              수원FC  
4              강원FC           Jeju United           제주SK FC

## Data Info and Basic Statistics


In [8]:
# Event data info
print("Event Data Info:")
print("=" * 50)
events.info()


Event Data Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 579306 entries, 0 to 579305
Data columns (total 18 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   game_id         579306 non-null  int64  
 1   action_id       579306 non-null  int64  
 2   period_id       579306 non-null  int64  
 3   time_seconds    579306 non-null  float64
 4   team_id         579306 non-null  int64  
 5   player_id       579305 non-null  float64
 6   result_name     229702 non-null  object 
 7   start_x         579306 non-null  float64
 8   start_y         579306 non-null  float64
 9   end_x           579306 non-null  float64
 10  end_y           579306 non-null  float64
 11  dx              579306 non-null  float64
 12  dy              579306 non-null  float64
 13  type_name       579306 non-null  object 
 14  player_name_ko  579305 non-null  object 
 15  team_name_ko    579306 non-null  object 
 16  position_name   509186 non-null  object

In [9]:
# Match info data info
print("Match Info Data Info:")
print("=" * 50)
match_info.info()


Match Info Data Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 198 entries, 0 to 197
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   game_id            198 non-null    int64 
 1   season_id          198 non-null    int64 
 2   competition_id     198 non-null    int64 
 3   game_day           198 non-null    int64 
 4   game_date          198 non-null    object
 5   home_team_id       198 non-null    int64 
 6   away_team_id       198 non-null    int64 
 7   home_score         198 non-null    int64 
 8   away_score         198 non-null    int64 
 9   venue              198 non-null    object
 10  competition_name   198 non-null    object
 11  country_name       198 non-null    object
 12  season_name        198 non-null    int64 
 13  home_team_name     198 non-null    object
 14  home_team_name_ko  198 non-null    object
 15  away_team_name     198 non-null    object
 16  away_team_name_ko  198

## Basic Sanity Checks


In [10]:
# Check coordinate ranges
# TODO: Adjust column names to match actual dataset
# Expected columns: start_x, start_y, end_x, end_y

if "start_x" in events.columns:
    print("Coordinate Ranges:")
    print(f"start_x: {events['start_x'].min():.2f} - {events['start_x'].max():.2f}")
    print(f"start_y: {events['start_y'].min():.2f} - {events['start_y'].max():.2f}")
    if "end_x" in events.columns:
        print(f"end_x: {events['end_x'].min():.2f} - {events['end_x'].max():.2f}")
        print(f"end_y: {events['end_y'].min():.2f} - {events['end_y'].max():.2f}")
else:
    print("TODO: Update column names for coordinates")


Coordinate Ranges:
start_x: 0.00 - 105.00
start_y: 0.00 - 68.00
end_x: 0.00 - 105.00
end_y: 0.00 - 68.00


In [11]:
# Distribution of event types
# TODO: Adjust column name to match actual dataset (e.g., "type_name")
if "type_name" in events.columns:
    print("Event Type Distribution:")
    print("=" * 50)
    event_counts = events["type_name"].value_counts()
    print(event_counts)
    print(f"\nTotal unique event types: {events['type_name'].nunique()}")
else:
    print("TODO: Update column name for event type")


Event Type Distribution:
type_name
Pass                      178582
Pass Received             167531
Carry                      88739
Recovery                   29713
Interception               13852
Out                        12136
Duel                       11244
Clearance                   9847
Tackle                      9498
Intervention                8835
Throw-In                    7461
Block                       6881
Cross                       6163
Shot                        4381
Pass_Freekick               4362
Foul                        4097
Goal Kick                   3030
Error                       2587
Ball Received               1743
Pass_Corner                 1729
Take-On                     1456
Catch                       1109
Parry                        807
Aerial Clearance             600
Hit                          581
Offside                      529
Goal                         515
Pause                        410
Defensive Line Support       320
Shot_Fre

In [12]:
# Distribution of results/outcomes
# TODO: Adjust column name to match actual dataset (e.g., "result_name")
if "result_name" in events.columns:
    print("Result Distribution:")
    print("=" * 50)
    result_counts = events["result_name"].value_counts()
    print(result_counts)
else:
    print("TODO: Update column name for result/outcome")


Result Distribution:
result_name
Successful            183372
Unsuccessful           41073
Off Target              1970
On Target               1117
Blocked                  859
Yellow_Card              640
Goal                     515
Low Quality Shot         105
Direct_Red_Card           22
Keeper Rush-Out           18
Second_Yellow_Card        11
Name: count, dtype: int64


## Notes / TODO

- [ ] Update column names in this notebook once exact schema is confirmed
- [ ] Check for missing values and decide on handling strategy
- [ ] Validate coordinate system (0-100 scale assumed)
- [ ] Identify any data quality issues (duplicates, outliers, etc.)
- [ ] Explore team names and IDs mapping


`00_data_overview.ipynb` 내용 확인 중입니다.


[1 tool called]


`00_data_overview.ipynb` 노트북 설명입니다.

## 📊 00_data_overview.ipynb - 데이터 개요 노트북

이 노트북은 원시 데이터를 탐색하고 기본 통계를 확인합니다.

---

### 📝 Cell 0: 목적 및 개요 (Markdown)

- 원시 이벤트 데이터와 경기 정보 구조 확인
- 좌표 범위, 이벤트 타입 분포 등 기본 검증
- 데이터 품질 이슈 확인

---

### 🔧 Cell 1: 환경 설정 및 임포트

```python
%load_ext autoreload
%autoreload 2  # 코드 변경 시 자동 리로드
```

- 프로젝트 루트를 `sys.path`에 추가
- 필요한 라이브러리 임포트 (pandas, numpy, matplotlib, seaborn)
- 프로젝트 모듈 임포트 (`data_loader`, `feature_engineering`)
- 시각화 스타일 설정

---

### 📥 Cell 2-4: 원시 데이터 로딩

#### Cell 3: 이벤트 데이터 로딩
```python
events = load_raw_events()
```

결과:
- 총 579,306개의 이벤트
- 18개 컬럼:
  - `game_id`, `action_id`, `period_id`, `time_seconds`
  - `team_id`, `player_id`
  - `start_x`, `start_y`, `end_x`, `end_y` (좌표)
  - `type_name` (이벤트 타입)
  - `result_name` (성공/실패)
  - `team_name_ko`, `player_name_ko` (한글 이름)

#### Cell 4: 경기 정보 로딩
```python
match_info = load_match_info()
```

결과:
- 총 198경기
- 17개 컬럼:
  - `game_id`, `home_team_id`, `away_team_id`
  - `home_score`, `away_score`
  - `home_team_name_ko`, `away_team_name_ko`
  - `venue`, `game_date` 등

---

### 📊 Cell 5-7: 데이터 정보 및 기본 통계

#### Cell 6: 이벤트 데이터 상세 정보
- 메모리 사용량: 약 79.6 MB
- 결측치:
  - `result_name`: 229,702개 (일부 이벤트는 결과 없음)
  - `player_id`: 1개
  - `position_name`: 70,120개

#### Cell 7: 경기 정보 상세 정보
- 메모리 사용량: 약 26.4 KB
- 결측치 없음

---

### ✅ Cell 8-11: 기본 검증 (Sanity Checks)

#### Cell 9: 좌표 범위 확인
```
start_x: 0.00 - 105.00
start_y: 0.00 - 68.00
end_x: 0.00 - 105.00
end_y: 0.00 - 68.00
```

참고: 실제 좌표는 0-100이 아니라 0-105 (x축), 0-68 (y축)입니다. 이는 실제 축구장 비율(105m × 68m)을 반영합니다.

#### Cell 10: 이벤트 타입 분포
총 37가지 이벤트 타입:
- Pass (178,582) - 가장 많음
- Pass Received (167,531)
- Carry (88,739)
- Shot (4,381)
- Goal (515)
- 기타: Tackle, Clearance, Cross 등

#### Cell 11: 결과(성공/실패) 분포
- Successful (183,372) - 성공
- Unsuccessful (41,073) - 실패
- Goal (515)
- On Target (1,117), Off Target (1,970)
- 기타: Yellow_Card, Blocked 등

---

### 📝 Cell 12: Notes / TODO

향후 작업:
- [ ] 컬럼명 확인 및 업데이트
- [ ] 결측치 처리 전략 수립
- [ ] 좌표 시스템 검증 (0-100 가정, 실제는 0-105/0-68)
- [ ] 중복, 이상치 확인
- [ ] 팀 이름과 ID 매핑 탐색

---

## 🔍 주요 발견 사항

1. 좌표 시스템: 0-100이 아니라 실제 필드 크기(105×68) 사용
2. 데이터 규모: 약 58만 개 이벤트, 198경기
3. 이벤트 타입: 37가지 (Pass가 가장 많음)
4. 결측치: `result_name`에 약 35만 개 결측 (일부 이벤트는 결과가 없음)

---

## 💡 다음 단계

1. 좌표 정규화: 0-100 스케일로 변환하거나, 현재 스케일(0-105/0-68) 그대로 사용
2. 결측치 처리: `result_name`이 없는 이벤트 처리 방법 결정
3. 다음 노트북: `01_pitch_utils_and_zones.ipynb`에서 피치 시각화 및 Zone 할당 테스트

이 노트북은 데이터의 첫 인상을 파악하는 데 사용됩니다.